dstep/translator/Record.d

```diff
@@ -59,9 +61,17 @@ private:
 
     string writeRecord (string name, void delegate (Data context) dg)
     {
+        auto known = cursor.definition in this.record_definitions;
+        if (known)
+            return "";
+
         auto context = new Data;
-        static if (is(typeof(context.isFwdDeclaration) == bool))
-            context.isFwdDeclaration = !cursor.isDefinition;
+        
+        if (cursor.isDefinition)
+            this.record_definitions[cursor] = true;
+        else
+            context.isFwdDeclaration = true;
+
```

Comment:
I think it would be better to handle all this in `Record.translate`. It's [these lines](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Record.d#L39-L42) that will write a struct declaration inline. When the definition is already know, don't execute those lines.